using-timescaledb/adaptive-chunking.md

```diff
@@ -0,0 +1,123 @@
+an existing hypertable, although this new setting only applies to
+chunks created after the change. A conflict resolution mechanism
+within TimescaleDB automatically detects any "collisions" between new
+and old chunks due to the change in partitioning, and cuts chunks to
+fit within existing constraints.
+
+*Adaptive chunking* leverages the ability to set new
+`chunk_time_interval`s, automating this process.  The system picks the
+first *open* (time) dimension to adapt, which is the same dimension
+that `chunk_time_interval` works on.  With adaptive chunking, however,
+the user instead sets a `chunk_target_size`, e.g., `1GB` and the
+system will try to adapt the `chunk_time_interval` on new chunks to
+reach this target chunk size using an algorithm.
+
+>:TIP: Setting a reasonable initial `chunk_time_interval` is still
+important because it allows the adaptive algorithm to quicker reach
+the target chunk size. Setting a too large `chunk_time_interval` is
+worse than setting a too small one, because the system will be
+committed to that larger interval until the next chunk. If no
+`chunk_time_interval` is set with adaptive chunking, the default is 1
+day.
+
+The default algorithm to calculate the next chunk's interval takes
+into account the *fill factor* of previous chunks, i.e., if they were
+over-filled or under-filled, and changes the interval accordingly. The
+system picks the first *open* (time) dimension to adapt. A user can
+set its own function to calculate the next `chunk_time_interval`,
+allowing customized algorithms.
+
+>:TIP: It is recommended that hypertables with adaptive chunking
+enabled has an index on the dimension being adapted, typically the
+"time" dimension. Such an index is created by default, unless
+otherwise specified. The index is used by the default adaptive
+chunking algorithm to get the `min` and `max` value from a chunk to
+calculate its fill factor. Without an index, it has to resort to a
+heap scan, which is expensive. The system will print `WARNING`s in the
+log if has to do heap scans.
+
+### Usage
+
+To create a hypertable that uses adaptive chunking, simply call
+[`create_hypertable`][1] as follows:
+
+```sql
+SELECT create_hypertable('conditions', 'time', chunk_target_size => '2GB');
+```
+
+This turns the table `conditions` into a hypertable that will try to
+adapt the time dimension to reach the given target chunk size.
+
+It is also possible to let the system *estimate* a target chunk size
+in case it is unclear what a good chunk size would be for the given
+system. In that case do:
+
+```sql
+SELECT create_hypertable('conditions', 'time', chunk_target_size => 'estimate');
+```
+
+This will estimate the chunk target size based on system settings and
+parameters.  In particular, it looks at the PostgreSQL configuration
+parameter for `effective_cache_size` as a measure of available disk
+cache. However, it will never go higher than system memory (actually
```

Comment:
Not sure what you are trying to convey here.  Are you saying that PostgreSQL __should__ go higher than available memory if `effective_cache_size` is configured properly?  Or are you saying that __even if__ it's configured improperly, the DB will still keep within those memory upper and lower bounds?

Reply:
I am trying to say something about how the estimate is made so that people know how to maybe affect it. But maybe I should just skip this, because estimate is probably for those that don't really care about internals or configuring other things.